test/extensions/filters/network/local_ratelimit/local_ratelimit_integration_test.cc

```diff
@@ -46,6 +48,83 @@ name: ratelimit
             test_server_->counter("local_rate_limit.local_rate_limit_stats.rate_limited")->value());
 }
 
+TEST_P(LocalRateLimitIntegrationTest, RateLimited) {
+  setup(R"EOF(
+name: ratelimit
+typed_config:
+  "@type": type.googleapis.com/envoy.extensions.filters.network.local_ratelimit.v3.LocalRateLimit
+  stat_prefix: local_rate_limit_stats
+  token_bucket:
+    max_tokens: 1
+    # Set fill_interval to effectively infinite so we only get max_tokens to start and never re-fill.
+    fill_interval: 100000s
+)EOF");
+
+  IntegrationTcpClientPtr tcp_client = makeTcpConnection(lookupPort("listener_0"));
+  FakeRawConnectionPtr fake_upstream_connection;
+  ASSERT_TRUE(fake_upstreams_[0]->waitForRawConnection(fake_upstream_connection));
+  ASSERT_TRUE(tcp_client->write("hello"));
+  ASSERT_TRUE(fake_upstream_connection->waitForData(5));
+  ASSERT_TRUE(fake_upstream_connection->write("world"));
+  tcp_client->waitForData("world");
+  tcp_client->close();
+  ASSERT_TRUE(fake_upstream_connection->waitForDisconnect());
+
+  tcp_client = makeTcpConnection(lookupPort("listener_0"));
+  tcp_client->waitForDisconnect();
+
+  EXPECT_EQ(1,
+            test_server_->counter("local_rate_limit.local_rate_limit_stats.rate_limited")->value());
+}
+
+TEST_P(LocalRateLimitIntegrationTest, SharedTokenBucket) {
+  config_helper_.addConfigModifier([&](envoy::config::bootstrap::v3::Bootstrap& bootstrap) -> void {
+    config_helper_.addNetworkFilter(R"EOF(
+name: ratelimit
+typed_config:
+  "@type": type.googleapis.com/envoy.extensions.filters.network.local_ratelimit.v3.LocalRateLimit
+  stat_prefix: local_rate_limit_stats
+  share_key: 'the_key'
+  token_bucket:
+    max_tokens: 2
+    # Set fill_interval to effectively infinite so we only get max_tokens to start and never re-fill.
+    fill_interval: 100000s
+)EOF");
+
+    // Clone the whole listener, which includes the `share_key`.
```

Comment:
Clone is not what the end-user does, is it better to add a new filter config to test?

Reply:
This is just an easy way to manipulate the protobufs in the test. It yields the same result as copy/paste the yaml to get the same fields.